src/Connectors/src/ConnectorBase/Relational/Oracle/OracleProviderConnectorOptions.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+
+namespace Steeltoe.CloudFoundry.Connector.Oracle
+{
+    public class OracleProviderConnectorOptions : AbstractServiceConnectorOptions
+    {
+        public const string Default_Server = "localhost";
+        public const int Default_Port = 1521;
+        private const string ORACLE_CLIENT_SECTION_PREFIX = "oracle:client";
+        private bool cloudFoundryConfigFound = false;
+
+        public OracleProviderConnectorOptions()
+        {
+        }
+
+        public OracleProviderConnectorOptions(IConfiguration config)
+            : base()
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+
+            var section = config.GetSection(ORACLE_CLIENT_SECTION_PREFIX);
+
+            section.Bind(this);
+
+            if (Uri != null)
+            {
+                Server = Uri.Split(':')[2].Substring(2);
+            }
+
+            Username = Uid;
```

Comment:
Is there a reason for including `Uid` and `Pw` and mapping them over to `Username` and `Password`?

For reference, this behavior is included with the SqlServer connector to account for the way the Sql Server Service Broker provides credentials. Is there a broker for Oracle?

Reply:
I think those properties were not removed when we reused the code from existing options class from other DB classes. I commented them and still able to connect to the Oracle DB. I will rerun all test cases and push the code shortly.

Reply:
Pushed the fixes for the review comment